docs(validations): add carbon validations and formik guide (#2986)
Sage/carbon: docs/validations.md

> # Getting started with validations, Formik and Yup

Reviewer: Could we just update this file `carbon/deploy/docs/guides/validations.md` with this information about `validation` instead od creating a new one ?

### Contents
* [Carbon validation overview](#carbon-validation-overview)
  * [Inputs](#inputs)
  * [Grouped inputs](#grouped-inputs)
  * [Validation icon placement](#validation-icon-placement)
* [Formik overview](#formik-overview)

## Carbon validation overview
Carbon provides inputs which are able to visually indicate three states of validations out of the box.

* **error**  –  meant to be a blocking validation, displays 2px wide red border and validation icon if message is provided.
* **warning**  –  meant to be a non-blocking validation, displays 1px wide orange border and validation icon if message is provided.
* **info**  –  meant to be a non-blocking validation, displays 1px wide blue border and validation icon if message is provided.

### Inputs

Each input component which supports validations accepts following props - `error`, `warning` or `info`.

> To display a border along with a validation icon and tooltip with a message pass a `string` as a content of one of these props.

Reviewer 2: ```suggestion
Passing a `string` to these props will display a border along with a validation icon and tooltip; this `string` value will be displayed as the tooltip message.
```

```js
<Textbox name="name" value={value} onChange={onChange} error="Message" />
```

> To display only a border without additional information why given validation status has occured simply pass one of validation props as a `boolean`.

Reviewer 2: ```suggestion
Passing a `boolean` to these props will display a border and validation icon without the additional tooltip information.
```

```js
<Textbox name="name" value={value} onChange={onChange} error={true} />
```

> Only exception from this is `DateRange` component which allows to display separate validation statuses on either `start` or `end` date  -  to achieve this `DateRange` component accepts `startError`, `endError`, `startWarning`, `endWarning`, `startInfo` and `endInfo` props. 

Reviewer 2: ```suggestion
The only exception from this is the `DateRange` component which displays separate validation statuses on either `start` or `end` dates  -  to achieve this the `DateRange` component accepts `startError`, `endError`, `startWarning`, `endWarning`, `startInfo` and `endInfo` props. 
```

```js
<DateRange name="name" value={value} onChange={onChange} startError="Start message" endError="End message" />
```

### Grouped inputs

Grouping components like `RadioButtonGroup`, `CheckboxGroup` and `ButtonToggleGroup` support validations too.

> Passing validation props on the group component will display validation icon on the group label/legend and apply proper styles to the grouped inputs.

Reviewer 2: ```suggestion
Passing validation props on a group component will display a validation icon on the group label/legend and apply proper styles to the grouped inputs.
```

> Interface is the same - it's up to the consumer to decide whether to display validation on group label/legend or on the individual component.

Reviewer 2: ```suggestion
The interface is the same - it's up to the consumer to decide whether to display validation on the group label/legend or on the individual component.
```

```js
<RadioButtonGroup legend="Legend" name="name" value={value} onChange={onChange} error="Validation on group legend">
  <RadioButton value="1" />
  <RadioButton value="2" />
</RadioButtonGroup>
```

```js
<RadioButtonGroup name="name" value={value} onChange={onChange}>
  <RadioButton value="1"  error="Validation on first grouped component" />
  <RadioButton value="2" />
</RadioButtonGroup>
```

### Validation icon placement

Most of the input components support two variants of validation icon placement:
* by the component or next to it - default behaviour
* next to the component label or legend

To decide which placement is preferred `validationOnLabel` or `validationOnLegend` props can be used.

### Form component

Carbon provides a Form component which augments an HTML `<form>` element with:

> * footer - container for rendering "Save", "Cancel" and other custom buttons, also might be sticky if wanted

Reviewer 2: ```suggestion
* footer - a container for rendering "Save", "Cancel" and other custom buttons; it can also be made sticky if wanted
```

controlling `value`/`checked` prop, and only supplying `name` `onChange` `onBlur` props), but this is strongly
discouraged :warning:

### Form validation using Formik

Formik supports three kinds of validation:

1. [**Form-level: plain**](https://jaredpalmer.com/formik/docs/guides/validation#validate)  –  A `values` object is
   passed to a `validate()` function which implements its own custom validation logic and returns an `errors` object.
2. [**Form-level: schema-based**](https://jaredpalmer.com/formik/docs/guides/validation#validationschema)  –  Performs
   validation using a [Yup](https://github.com/jquense/yup) schema.
3. [**Field-level**](https://jaredpalmer.com/formik/docs/guides/validation#field-level-validation)  –  Separate
   `validate()` functions are supplied for each form field.  This is the closest equivalent functionality of Carbon's
   [`withValidation`](src/components/validations/with-validation.hoc.js) HOC which attaches validation functions to an
   individual form field.

## Code examples

### [Form validation using Field-level validation, Yup as a schema builder and hooks](https://codesandbox.io/s/carbon-formik-yup-y6pzt)

> 

Reviewer: and remove all those empty lines from here :)
